source/h2_connection.c

```diff
@@ -1563,9 +1530,28 @@ static void s_handler_installed(struct aws_channel_handler *handler, struct aws_
             goto error;
         }
     }
-    /* Initial settings are already enqueued */
-    aws_h2_try_write_outgoing_frames(connection);
+    struct aws_h2_pending_settings *init_pending_settings = connection->thread_data.init_pending_settings;
+    /* Set user_data here, the user_data is valid now */
+    init_pending_settings->user_data = connection->base.user_data;
```

Comment:
We could do the following here:
```
aws_linked_list_push_back(&connection->thread_data.pending_settings_queue, init_pending_settings);
connection->thread_data.init_pending_settings = NULL;
```

and then we could remove the special-case code for `connection->thread_data.init_pending_settings` from `s_decoder_on_settings_ack()` and `s_finish_shutdown()`